src/App.jsx

```diff
@@ -41,9 +124,19 @@ class App extends Component {
     firebaseWrapper.logout();
   };
 
-  render() {
-    const { userId } = this.state;
+  makeNotification = (text, color, time, image) => {
+    const { notifications } = this.state;
```

Comment:
when creating a new state based off of the previous state/props, you should use this version of setState
```
this.setState((state, props) => {
  return {counter: state.counter + props.step};
});
```

Reply:
notification panel wouldn't auto update with the notification when I did it this way. Left the code in a comment if you want that